File: src/main/java/axirassa/model/UserEmailAddressEntity.java

package axirassa.model;

import javax.persistence.Basic;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;

import lombok.Getter;
import lombok.Setter;

import org.hibernate.annotations.Cache;
import org.hibernate.annotations.CacheConcurrencyStrategy;
import org.hibernate.annotations.NaturalId;

import axirassa.model.interceptor.EntityPreSave;
import axirassa.util.AutoSerializingObject;
import axirassa.util.RandomStringGenerator;

@Entity
@Cache(usage = CacheConcurrencyStrategy.READ_WRITE)
public class UserEmailAddressEntity extends AutoSerializingObject implements EntityPreSave, EntityWithUser {
	private static final long serialVersionUID = 4776151084882073597L;

	//
	// Instance
	//
	@Id
	@Getter
	@Setter
	@Basic(optional = false)
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Long id;

	@Getter
	@Setter
	@Basic(optional = false)
	private boolean primaryEmail;

	@Getter
	@Setter
	@ManyToOne(optional = false)
	private UserEntity user;

	@NaturalId
	@Getter
	@Setter
	@Basic(optional = false)
	private String email;

	@Getter
	@Setter
	@Basic(optional = false)
	private String token;

	@Getter
	@Setter
	@Basic(optional = false)
	private boolean verified;


	@Override
	public void preSave() {
		if (token == null)
			token = RandomStringGenerator.makeRandomStringToken(36);
	}
}

File: src/main/java/axirassa/model/UserEmailAddressEntity.scala
package axirassa.model

import javax.persistence.Basic
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import scala.reflect.BeanProperty
import org.hibernate.annotations.Cache
import org.hibernate.annotations.CacheConcurrencyStrategy
import javax.persistence.ManyToOne
import axirassa.util.AutoSerializingObject
import axirassa.model.interceptor.EntityPreSave
import axirassa.util.RandomStringGenerator
import org.hibernate.annotations.NaturalId

@Entity
@Cache(usage = CacheConcurrencyStrategy.READ_WRITE)
class UserEmailAddressEntity extends AutoSerializingObject with EntityPreSave with EntityWithUser {
	//
	// Instance
	//
	@Id
	@BeanProperty
	@Basic(optional = false)
	@GeneratedValue(strategy = GenerationType.AUTO)
	var id: Long = _


	@BeanProperty
	@Basic(optional = false)
	var primaryEmail: Boolean = _


	@BeanProperty
	@ManyToOne(optional = false)
	var user: UserEntity = _



	@NaturalId
	@BeanProperty
	@Basic(optional = false)
	var email: String = _


	@BeanProperty
	@Basic(optional = false)
	var token: String = _

	@BeanProperty
	@Basic(optional = false)
	var verified: Boolean = _


	override
	def preSave () {
		if (token == null)
			token = RandomStringGenerator.makeRandomStringToken(36)
	}
}